app/app.ts

```diff
@@ -396,32 +405,77 @@ function setupIpc(localSettings: LocalSettings, scrSettings: ScrSettings) {
   )
 
   ipcMain.handle('fsReadFile', async (_, filePath) => {
-    return fsPromises.readFile(filePath)
+    try {
+      const arrayBuffer = await fsPromises.readFile(filePath)
+      const result: FsFile = {
+        file: arrayBuffer,
+        error: false,
+      }
+      return result
+    } catch (e) {
+      if ((e as any).code === 'ENOENT') {
+        const error: FsError = {
+          error: true,
+          code: FsErrorCode.ENOENT,
+        }
+        return error
+      }
+      throw e
+    }
   })
   ipcMain.handle('fsReadDir', async (_, dirPath, options) => {
-    const result = await fsPromises.readdir(dirPath, options)
-    return result.map<FsDirent>(d => ({
-      isFile: d.isFile(),
-      isDirectory: d.isDirectory(),
-      name: d.name,
-    }))
+    try {
+      const result = await fsPromises.readdir(dirPath, options)
+      const files = result.map<FsDirent>(d => ({
+        isFile: d.isFile(),
+        isDirectory: d.isDirectory(),
+        name: d.name,
+      }))
+      const dirents: FsDirents = {
+        files,
+        error: false,
+      }
+      return dirents
+    } catch (e) {
+      if ((e as any).code === 'ENOENT') {
+        const error: FsError = {
+          code: FsErrorCode.ENOENT,
+          error: true,
+        }
+        return error
+      }
+      throw e
+    }
   })
   ipcMain.handle('fsStat', async (_, filePath) => {
-    const result = await fsPromises.stat(filePath)
-    return {
-      isFile: result.isFile(),
-      isDirectory: result.isDirectory(),
-      size: result.size,
-      blksize: result.blksize,
-      blocks: result.blocks,
-      atimeMs: result.atimeMs,
-      mtimeMs: result.mtimeMs,
-      ctimeMs: result.ctimeMs,
-      birthtimeMs: result.birthtimeMs,
-      atime: result.atime,
-      mtime: result.mtime,
-      ctime: result.ctime,
-      birthtime: result.birthtime,
+    try {
+      const result = await fsPromises.stat(filePath)
+      const stat: FsStats = {
+        isFile: result.isFile(),
+        isDirectory: result.isDirectory(),
+        size: result.size,
+        blksize: result.blksize,
+        blocks: result.blocks,
+        atimeMs: result.atimeMs,
+        mtimeMs: result.mtimeMs,
+        ctimeMs: result.ctimeMs,
+        birthtimeMs: result.birthtimeMs,
+        atime: result.atime,
+        mtime: result.mtime,
+        ctime: result.ctime,
+        birthtime: result.birthtime,
+        error: false,
+      }
+      return stat
+    } catch (e) {
+      if ((e as any).code === 'ENOENT') {
+        const error: FsError = {
```

Comment:
You can just return the object directly instead of creating a dedicated variable for it, e.g.
```ts
return {
  error: true,
  code: FsErrorCode.FilepathNotFound,
}
```

Reply:
Ah ok, sorry I assumed that resolved comments means they're handled and pushed.